### PR TITLE
Add alternate implementation for macOS 12.0+.

### PR DESCRIPTION
**Pull request by dkocher**

### CODE REVIEW

File: core/dylib/src/main/java/ch/cyberduck/core/preferences/ApplicationPreferences.java [hunk 1]
```diff
@@ -84,7 +86,13 @@ protected void setFactories() {
         this.setDefault("factory.notification.class", NotificationCenter.class.getName());
         this.setDefault("factory.iconservice.class", WorkspaceIconService.class.getName());
         this.setDefault("factory.filedescriptor.class", LaunchServicesFileDescriptor.class.getName());
-        this.setDefault("factory.schemehandler.class", LaunchServicesSchemeHandler.class.getName());
+        if(Factory.Platform.osversion.matches("10\\.(12|13|14|15|16).*")) {
```
Comment (ylangisc) [inline core/dylib/src/main/java/ch/cyberduck/core/preferences/ApplicationPreferences.java:89]: What about Big Sur (11.x)? It seems that it is correctly reported as of JDK 17, refer to https://bugs.openjdk.org/browse/JDK-8253702 and https://github.com/openjdk/jdk/commit/66757750

Comment (dkocher) [reply core/dylib/src/main/java/ch/cyberduck/core/preferences/ApplicationPreferences.java:89]: Test the assumption in #13540.

**Review by ylangisc — CHANGES_REQUESTED** [review]
Hanging tests on macOS 12.

**Comment by dkocher** [comment]
> Hanging tests on macOS 12.

Unfortunately cannot be reproduced locally.

**Comment by ylangisc** [comment]
> > Hanging tests on macOS 12.
> 
> Unfortunately cannot be reproduced locally.

Related to any headless setup of the CI env?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
         strategy:
             fail-fast: false
             matrix:
-                os: [ macos-10.15, macos-11, macos-12, ubuntu-latest, windows-latest ]
+                os: [ macos-11, macos-12, ubuntu-latest, windows-latest ]
```
Comment (ylangisc) [inline .github/workflows/build.yml:16]: Why macOS 10.15 has been removed?

Comment (dkocher) [reply .github/workflows/build.yml:16]: > GitHub Actions The macOS 10.15 Actions runner image is being deprecated and will be removed by 8/30/22

**Review by ylangisc — APPROVED** [review]

**Pull request merged**